backend/Tim.Backend/Startup/ServiceExtensions.cs

```diff
@@ -137,85 +135,136 @@ public static void AddInstrumentationServices(this IServiceCollection services,
-            var dbConfigs = new DatabaseClasses();
-            var kustoConfigs = configuration.GetSection(nameof(KustoConfiguration)).Get<KustoConfiguration>() ?? new KustoConfiguration();
             var authConfigs = configuration.GetSection(nameof(AuthConfiguration)).Get<AuthConfiguration>() ?? new AuthConfiguration();
-
-            secrets.Validate();
-            kustoConfigs.Validate();
             authConfigs.Validate();
 
-            services.AddSingleton(p =>
-            {
-                var connectionString = new KustoConnectionStringBuilder
-                {
-                    DataSource = kustoConfigs.IngestKustoClusterUri,
-                    InitialCatalog = kustoConfigs.KustoDatabase,
-                    FederatedSecurity = true,
-                    ApplicationClientId = kustoConfigs.KustoAppId,
-                    ApplicationKey = secrets.KustoAppKey,
-                    ApplicationNameForTracing = "TIM-Ingest-To-Kusto",
-                };
-
-                return new KustoIngestClient(connectionString);
-            });
-
-            services.AddSingleton(p =>
-            {
-                var connectionString = new KustoConnectionStringBuilder
-                {
-                    DataSource = kustoConfigs.KustoClusterUri,
-                    InitialCatalog = kustoConfigs.KustoDatabase,
-                    FederatedSecurity = true,
-                    ApplicationClientId = kustoConfigs.KustoAppId,
-                    ApplicationKey = secrets.KustoAppKey,
-                    ApplicationNameForTracing = "TIM-Query-To-Kusto",
-                };
-
-                return new KustoQueryClient(connectionString);
-            });
-
             var authenticationSchemes = new List<string>();
 
             services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
                 .AddJwtBearer(options =>
                 {
+                    options.Audience = authConfigs.ClientId;
+                    options.Authority = $"https://login.microsoftonline.com/{authConfigs.ClientAuthority}";
                     options.TokenValidationParameters = new TokenValidationParameters
                     {
-                        ValidateIssuer = false,
-                        ValidateAudience = false,
+                        ValidateIssuer = true,
+                        ValidateAudience = true,
                         ValidateIssuerSigningKey = true,
-                        IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(authConfigs.SigningKey)),
+                        /*IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(authConfigs.SigningKey))*/
+                        ValidIssuer = "https://login.microsoftonline.com/{authConfigs.ClientAuthority}",
+                        ValidAudience = $"api://{authConfigs.ClientId}",
```

Comment:
so did you settle on the fact that we should you an azure app after all? or am I reading this wrong?

Reply:
Pretty much.
As long as we use Kusto, we'll need OBO it seems and therefore, some kind of Azure authentication. Once I start introducing non Kusto query platforms, we can make this configurable.